internal/resources/grafana/resource_alerting_rule_group.go

```diff
@@ -706,3 +733,36 @@ func unpackNotificationSettings(p interface{}) (*models.AlertRuleNotificationSet
 	}
 	return &result, nil
 }
+
+func packRecord(r *models.Record) interface{} {
+	if r == nil {
+		return nil
+	}
+	res := map[string]interface{}{}
+	if r.Metric != nil {
+		res["metric"] = *r.Metric
+	}
+	if r.From != nil {
+		res["from"] = *r.From
+	}
+	return []interface{}{res}
+}
+
+func unpackRecord(p interface{}) *models.Record {
+	if p == nil {
+		return nil
+	}
+	list := p.([]interface{})
+	if len(list) == 0 {
+		return nil
+	}
```

Comment:
Maybe add safe check here?
```suggestion
	list, ok := p.([]interface{})
	if !ok || len(list) == 0 {
		return nil
	}
```